test/extended/etcd/helpers/client.go

```diff
@@ -0,0 +1,117 @@
+package helpers
+
+import (
+	"bufio"
+	"context"
+	"fmt"
+	"os/exec"
+	"strconv"
+	"strings"
+	"time"
+
+	clientv3 "go.etcd.io/etcd/client/v3"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	restclient "k8s.io/client-go/rest"
+)
+
+type EtcdClientCreator interface {
```

Comment:
How about doing something similar to Prometheus Client from https://github.com/openshift/cluster-kube-controller-manager-operator/pull/623 to speak directly with etcd rather than relying on port-forward?

Reply:
I don't think so, the difference is that the test doesn't run inside the cluster. It needs to access the etcd cluster from the public internet.